docs/design-proposals/apiserver/openstack-api-support.md

```diff
@@ -0,0 +1,408 @@
+---
+title: Support Openstack Compute APIs in Arktos
+authors:
+  - "@yb01"
+---
+
+# Support Openstack Compute APIs in Arktos
+## Background and context
+To boost Arktos VM adoption and to let Openstack users to easily explore Arktos VM support without a lot of changes from the users’ applications, it is decided to add a new set of APIs in Arktos to match Openstack Server related operations. Besides better user experiences, this new APIs will also help existing Openstack perf tests to be relatively easily converted to run against Arktos.
+
+## Goal and non-goals
+
+Goals:
+
+1. New set of REST interfaces in Arktos to support Openstack requests for Server CRUD
+1. New set of REST interfaces in Arktos to support Openstack requests for Server Actions
+1. Batch support with Arktos deployments
+
+Non-goals:
+
+1. Nova or Nova test tool can switch to Arktos for basic VM operation, by switching to the Arktos URL. 
+1. Openstack client-side SDK or other dev tools are not in the scope of this design doc.
+1. Openstack API other than VM CRUD, VM Actions, is not in the scope of this design doc.
+
+## New APIs in Arktos
+### New API route path:
+The new route for servers
+
+* 	{Arktos-service-url}/servers
+* 	{Arktos-service-url}/servers/{server-id}/detail
+
+The new route for server actions
+
+* {Arktos-service-url}/servers/{server-id}/action
+
+
+### API call chain
+The API call chains of /servers will be the same as the other API requests, which is created in the DefaultBuildHandlerChain() function
+
+
+### VM list, create, delete
+
+|Methods |URL.     |Function                     |Supported in Arktos|
+|--------|---------|-----------------------------|-------------------|
+|GET	  |/servers  |	List all VMs for a tenant/namespace | yes |
+|POST	  |/servers	 |Create a VM	                |yes |
+|POST	  |/servers	 |Create multiple VMs.        |Not for this release. Consider this after the initial API support. Arktos can achieve this by calling the deployment API internally with replicaset of VM pods [future works](Create VMs in batch)|
+|GET	  |/servers/detail	| List VM details for a tenant/namespace | Not for this release|
```

Comment:
Please add some note at the beginning and explain what is "this release"

Reply:
Fixed all to explicitly use Arktos 130 release term